### PR TITLE
Revert "Revert "Do not remove server constants from GROUP BY key for secondary query.""

### PR DESCRIPTION
**Pull request by KochetovNicolai**
Reverts ClickHouse/ClickHouse#63297

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 9d55bc82d6609633a135a5044f05aadeaff21755 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/63415/9d55bc82d6609633a135a5044f05aadeaff21755/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by novikd — APPROVED** [review]
In general LGTM

File: src/Planner/PlannerExpressionAnalysis.cpp [hunk 1]
```diff
@@ -51,6 +51,41 @@ FilterAnalysisResult analyzeFilter(const QueryTreeNodePtr & filter_expression_no
...
+        const auto * constant_node = node->as<ConstantNode>();
+        const auto * function_node = node->as<FunctionNode>();
+        if (constant_node)
+        {
+            if (!isDeterministicConstant(*constant_node))
+                return false;
+        }
+        else if (function_node)
+        {
+            if (!function_node->getFunctionOrThrow()->isDeterministic())
```
[the remark this answers is not in the document]
Comment (KochetovNicolai) [reply src/Planner/PlannerExpressionAnalysis.cpp:76]: I know that server constants are deterministic, but not in scope of query, so isDeterministicInScopeOfQuery is not enough :)

**Comment by KochetovNicolai** [comment]
02455_one_row_from_csv_memory_usage is flaky, 
AST fuzzer (tsan) - is known issue with prewhere

**Pull request merged**